src/components/formio/radio.stories.ts

```diff
@@ -88,3 +88,36 @@ export const WithErrors: Story = {
     await expect(canvas.queryByText('Example error')).toBeInTheDocument();
   },
 };
+
+export const DefaultValueRemoved: Story = {
+  args: {
+    label: 'With removing the default value',
+    options: [
+      {value: 'a', label: 'A'},
+      {value: 'b', label: 'B'},
+    ],
+  },
+
+  parameters: {
+    formik: {
+      initialValues: {'my-radio': 'a'},
+    },
+  },
+
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+    const input1 = canvas.getByLabelText('A');
+    const input2 = canvas.getByLabelText('B');
+
+    // Before clearing the default value
+    await expect(canvas.queryByText('Clear selection')).toBeInTheDocument();
+    await expect(input1).toBeChecked();
+    await expect(input2).not.toBeChecked();
+
+    // After clearing the default value
+    await userEvent.click(canvas.getByRole('button', {name: 'Clear selection'}));
+    await expect(canvas.queryByText('Clear selection')).not.toBeInTheDocument();
```

Comment:
perhaps you can re-use the button reference from earlier when you assert that it's present, that way the test is more robust in case the literal text changes:

```tsx
const button = canvas.getByRole('button', {name: 'Clear selection'});
expect(button).toBeVisible();
await userEvent.click(button);
expect(button).not.toBeInTheDocument();
```

I'm not sure this works though, but please give it a try.